tutorpicasso/commands/enable_private_packages.py

```diff
@@ -0,0 +1,115 @@
+import os
+import subprocess
+from typing import Any, Dict
+
+import click
+from packaging.version import Version
+from tutor import config as tutor_config
+from tutor.__about__ import __version__ as tutor_version
+
+
+@click.command(name="enable-private-packages", help="Enable picasso private packages")
+def enable_private_packages() -> None:
+    """
+    Enable private packages command.
+
+    This command enables picasso private packages by cloning the packages and
+    defining them as private.
+
+    Raises:
+        Exception: If there is not enough information to clone the repo.
+    """
+    context = click.get_current_context().obj
+    tutor_root = context.root
+    tutor_conf = tutor_config.load(tutor_root)
+
+    tutor_version_obj = Version(tutor_version)
+    # Define Quince version as the method for installing private packages changes from this version
+    quince_version_obj = Version("v17.0.0")
+
+    # Use these specific paths as required by Tutor < Quince
+    private_requirements_root = f"{tutor_root}/env/build/openedx/requirements"
+    packages = get_picasso_packages(tutor_conf)
+
+    # Create necessary directory if it doesn't exist
+    if not os.path.exists(private_requirements_root):
+        os.makedirs(private_requirements_root)
+
+    for package, info in packages.items():
+        if not {"name", "repo", "version"}.issubset(info):
+            raise click.ClickException(
+                f"{package} is missing one of the required keys: 'name', 'repo', 'version'"
+            )
+
+        if os.path.isdir(f'{private_requirements_root}/{info["name"]}'):
+            subprocess.call(
+                ["rm", "-rf", f'{private_requirements_root}/{info["name"]}']
+            )
+
+        subprocess.call(
+            ["git", "clone", "-b", info["version"], info["repo"]],
+            cwd=private_requirements_root,
+        )
+
+        if tutor_version_obj < quince_version_obj:
+            private_requirements_txt = f"{private_requirements_root}/private.txt"
+            _enable_private_packages_before_quince(info, private_requirements_txt)
+        else:
+            _enable_private_packages(info, private_requirements_root)
```

Comment:
[nit] We're using both packages and requirements in naming. Could we decide whether to use packages or requirements?

Reply:
Thanks a lot. I applied https://github.com/eduNEXT/tutor-contrib-picasso/commit/7c6ccc3dc1af7095e26cb2f1f7829ce20a303a17 and also refactored a little bit the names of the functions.

Regarding naming, I propose that in the context of Picasso, we use the name package (to read the package information and enable packages, but for the context of Tutor, we use the name requirements); I don't know if there is a good place to put it. 

I maintained the package command to avoid being so far to `tutor-contrib-edunext-distro`; but later, we can question the use of package instead of requirements.